### ISSUE TITLE
model.move() not moving non-parameter tensors

### ISSUE THREAD

**Issue by dsyme** [DiffSharp/DiffSharp#257]
We have `model.move(device)` and this iterates all parameter tensors and moves them

However there are sometimes non-parameter tensors like `momentum` in BatchNorm2d and these are of course not being moved currently.

```fsharp
type BatchNorm2d(numFeatures:Int, ?eps:double, ?momentum:Tensor, ?affine:bool, ?trackRunningStats:bool, ?reversible:bool) =
    inherit Model()
    let eps = defaultArg eps 1e-5
    let momentum = defaultArg momentum (dsharp.tensor(0.1))
    ...
    let w = Parameter <| if affine then dsharp.ones(numFeatures) else dsharp.zero() // gamma
    let b = Parameter <| if affine then dsharp.zeros(numFeatures) else dsharp.zero() // beta
    let _mean = Parameter <| dsharp.zero()
    let _variance = Parameter <| dsharp.zero()
    ...
```

If we decide we want to continue to support `move` then I'd imagine we should put these in a `Parameter` holder marked with a flag that they are not differentiable hence don't contribute to `model.parameters`

Separately I notice that the order of the `parameters` and `subModels` is somewhat randomly determined by the order of entries in the Dictionary objects.  We should always return these in a consistent order.

**Comment by VisualMelon** [comment]
It would be useful for parameter updatability to be dynamic (i.e. the flag is variable in some parameter type): in some models, one may want to update only a subset of model parameters in different parts of the system. (I've done this previously, but performing the updates manually). I'd prefer such things did appear in `parameters` (is this not necessary for save/load?), but I'm not sure how much of a breaking change that would be (optim would be fine, not checked `reverseDiff` or if there are other spots that would need to know about it).

**Comment by gbaydin** [comment]
This is addressed in #381 which introduces improvements in model implementation and makes the parameter and sub-model order deterministic and reliable.
